TestFoundation/TestFileManager.swift

```diff
@@ -442,8 +442,8 @@ class TestFileManager : XCTestCase {
         }
         
         do {
-            let _ = try fm.contentsOfDirectory(atPath: "")
-            
+            // Check a bad path fails
```

Comment:
Why the change here, for the record?

Reply:
Its for compatibility with Darwin's native Foundation. This call will throw an exception under native Foundation as the filename is passed through `fileSystemRepresentation(withPath:)` which treats `nil` and `""` as invalid paths. This PR modifies `contentsOfDirectory(atPath:)` and `subpathsOfDirectory(atPath:)` to now also pass the path through this function so `""` is now going to fail the `precondition(path != "", "Empty path argument")` check.

FYI, if you open the Xcode project `DarwinCompatibilityTests` in the `swift-corelibs-foundation` directory you can run all of the tests as unit tests against native Foundation to see what differences currently exist.